doc/man3/ASYNC_WAIT_CTX_new.pod

```diff
@@ -103,14 +120,56 @@ code that the job should be resumed by making the wait file descriptor
 "readable". Once resumed the engine should clear the wake signal on the wait
 file descriptor.
 
+As well as a file descriptor, user code may also be notified via a callback. The
+callback and data pointers are stored within the ASYNC_WAIT_CTX along with an
+additional status field that can be used for the notification of retries from an
+engine. This additional method can be used when the user thinks the a file
+descriptor is too costly in terms of CPU cycles or in some context where a file
+descriptor is not appropriate.
+
+ASYNC_WAIT_CTX_set_callback() sets the callback and the callback argument. The
+callback will be called to notify user code when an engine completes a
+cryptography operation.
+
+ASYNC_WAIT_CTX_get_callback() returns the callback set in the ASYNC_WAIT_CTX
+structure.
+
+ASYNC_WAIT_CTX_set_status() allows an engine to set the current engine status.
+The possible status values are the following:
+ASYNC_STATUS_UNSUPPORTED: The engine does not support the callback mechanism.
+This is the default value.
+ASYNC_STATUS_ERR: The engine has a fatal problem with this request. The user
+code should clean up this session.
```

Comment:
So, if the engine sets the status to this value - would it call ASYNC_pause_job() or not? Or would it just return with an error (i.e. so that the job finishes with ASYNC_FINISH, and "ret" indicates an error). If the latter, then I'm not sure what this status means? The status is set on the ASYNC_WAIT_CTX, but the job is not waiting.

Reply:
I think both cases are okay to handle, and we'd rather to keep this pre-define value here in case someone wants to use it.